x-pack/plugins/aiops/public/hooks/use_data.ts

```diff
@@ -54,9 +54,14 @@ export const useData = (
     timeRangeSelector: selectedDataView?.timeFieldName !== undefined,
     autoRefreshSelector: true,
   });
+  const timeRangeMemoized = useMemo(
```

Comment:
Nit: not sure the useMemo is necessary here as calling `getActiveBounds` is needed anyway to see if the active bounds have changed and that's all this is doing.

Reply:
In this case this is not about performance but to avoid render loops. `timeRange` is a nested object and we need to avoid getting a new instance with the same values, that's why it uses `JSON.stringify` in the comparator to get the same instance if no values changed. Otherwise it would trigger the other `useEffect` after this to be called with every new instance.